docs/recipes/travis.md

```diff
@@ -0,0 +1,151 @@
+
+```yaml
+language: node_js
+
+node_js:
+  - 8
+  - 6
+  - 4
+
+os:
+  - linux
+  - osx
+
+script:
+  # Run tests
+  - npm run test
+
+  # Only for a local semantic-release installation
+  - npm run travis-deploy-once "npm run semantic-release"
+
+  # Only for a global semantic-release installation
+  - npm install -g travis-deploy-once semantic-release
+  - travis-deploy-once "semantic-release"
+```
+
+**Note**: See the `travis-deploy-once` [`pro`](https://github.com/semantic-release/travis-deploy-once#-p---pro) and [`travis-url`](https://github.com/semantic-release/travis-deploy-once#-u---travis-url) options for using with [Travis Pro](https://docs.travis-ci.com/user/travis-pro) and [Travis Enterprise](https://enterprise.travis-ci.com).
+
+### `package.json` configuration for multiple Node jobs
+
+A `package.json` is required only for [local](../usage/installation.md#local-installation) **semantic-release** installation.
+
+```json
+{
+  "devDependencies": {
+    "semantic-release": "^12.0.0",
+    "travis-deploy-once": "^4.0.0"
+  },
+  "scripts": {
+    "semantic-release": "semantic-release",
+    "travis-deploy-once": "travis-deploy-once"
+  }
+}
+```
+
+## Non-JavaScript projects configuration
+
+For projects that require to be tested with one or multiple version of a Non-JavaScript [language](https://docs.travis-ci.com/user/languages), optionally on multiple [Operating Systems](https://docs.travis-ci.com/user/multi-os).
+
+This recipe cover the Travis specifics only. See [Non JavaScript projects recipe](../support/FAQ.md#can-i-use-semantic-release-to-publish-non-javascript-packages) for more information on the **semantic-release** configuration.
+
+### `.travis.yml` configuration for non-JavaScript projects
+
+This example is a minimal configuration for semantic-release with a build running [Go 1.6 and 1.7](https://docs.travis-ci.com/user/languages/go) on Linux and OSX. See [Travis - Customizing the Build](https://docs.travis-ci.com/user/customizing-the-build) for additional configuration options.
+
+This example uses [`travis-deploy-once`](https://github.com/semantic-release/travis-deploy-once) in order to [run `semantic-release` only after all tests succeeded](../usage/ci-configuration.md#run-semantic-release-only-after-all-tests-succeeded). Alternatively you can use [Travis CI Build Stages recipe](travis-build-stages.md).
+
+It's recommended to run the `semantic-release` command in the [Travis `script` step](https://docs.travis-ci.com/user/customizing-the-build/#The-Build-Lifecycle) so if an error happen the build will fail and Travis will send a notification.
+
+```yaml
+language: go
+
+go:
+  - 1.6
+  - 1.7
+
+os:
+  - linux
+  - osx
+
+script:
+  # Run tests
+  - go test -v ./...
+
+  # Use nvm to install and use the Node LTS version (nvm is installed on Travis images)
+  - nvm install lts/*
```

Comment:
I would always pin this to a specific major version, e.g. 8. Otherwise if a new major version becomes lts this would break the build.

Reply:
We are guaranteeing the support of Node LTS whatever version it is. See [Node Support Policy](https://github.com/semantic-release/semantic-release/blob/caribou/docs/support/node-support-policy.md).

Reply:
Sure, the _latest_ version of semantic-release guarantees that. But if a project is on an older pinned major version of semantic-release (which it should be for reproducable builds) then a new LTS release can break the build.

Reply:
The doc clearly mention is a simplified example.

An example cannot plan for every possible interaction in every situation otherwise it ends up being confusing with a lot of "If this do that".

The person configuring the `.travis.yml` file expected to have a minimum understanding of Travis. The objective of this doc is explain how to use semantic-release, not to explain how to use Travis, or to choose which Node version is best for your project or your company.

The users who configure the `.travis.yml` should be able to analyze by themselve the fact they specifically pin a very old version of semantic-release.